### PR TITLE
Event Items Framework

### PR DESCRIPTION
**Pull request by Geevies**
Made a framework for adding event items to the loadout, but restricting it only to character that could feasibly have been at the event. This is done by adding the date the event was ran into the loadout item, meaning only character created before that date has that item available in the loadout.

image

Also adds an optional header, which you can use to name the group of items, with the event's name or whatever you want.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Review by alsoandanswer — APPROVED** [review]
This will unironically solve people deleting their characters to circumvent things

...however may or may not encourage elitism. May be just me though.

File: code/modules/client/preference_setup/loadout/loadout.dm [hunk 1]
```diff
@@ -275,12 +283,15 @@ var/list/gear_datums = list()
 				temp_html += " <a href='?src=\ref[src];gear=[G.display_name];tweak=\ref[tweak]'>[tweak.get_contents(get_tweak_metadata(G, tweak))]</a>"
 			temp_html += "</td></tr>"
 
-		if(ticked)
-			ticked_items_html += temp_html
-		else if(!available)
+		if(G.always_use_available_items_list)
```
Comment (DreamySkrell) [inline code/modules/client/preference_setup/loadout/loadout.dm:286]: I don't like the headers and this, in how it circumvents the existing system of {selected items at the top}, {available items in the middle}, and {unavailable at the bottom}.

Meaning that items that are `always_use_available_items_list` will always be at the top, even if they can't be selected and are unavailable. Potentially makes the loadout even more of a mess than it is currently.

**Comment by KingOfThePing** [comment]
I do not understand this PR

File: code/modules/client/preference_setup/loadout/loadout.dm (hunk 1 above)
Comment (DreamySkrell) [reply code/modules/client/preference_setup/loadout/loadout.dm:286]: What would the headers be even used for, that can't already be done with adding a new category.

**Comment by DreamySkrell** [comment]
Other than the code review above, I don't like this, cause elitism and gatekeeping and whatever else you can call it. I'd be against items from events altogether, personally. Like, introducing new chars is already hard enough, and event-locked drip is not going to help with that.

> "you're not part of the cool club cause you weren't here 5 years ago, and did not take part in these cool events"

**Comment by Arrow768** [comment]
I dont see the point of that.
Imho that is a suboptimal implementation as it allows every character that existed before a specific point to use the item (no matter if they were present during the event or not)

In addition, we already have a more advanced version of that capability.
We have a DB table that tracks with characters were present during which rounds.
The custom items are stored in the DB aswell.
Its just a question of querying the characters that were present during round x (or have been played in a specific time range) and add the "event item" to their custom item list.

It might be useful to be able to set custom items as "available in loadout" instead of always spawning so players can decide if they want to spawn with them.

**Pull request closed without merging**